forgerock-auth/src/test/java/org/forgerock/android/auth/callback/StringAttributeInputCallbackTest.java

```diff
@@ -0,0 +1,115 @@
+                "                                        \"string\"\n" +
+                "                                    ]\n" +
+                "                                }\n" +
+                "                            },\n" +
+                "                            {\n" +
+                "                                \"policyId\": \"minimum-length\",\n" +
+                "                                \"params\": {\n" +
+                "                                    \"minLength\": 1\n" +
+                "                                },\n" +
+                "                                \"policyRequirements\": [\n" +
+                "                                    \"MIN_LENGTH\"\n" +
+                "                                ]\n" +
+                "                            },\n" +
+                "                            {\n" +
+                "                                \"policyId\": \"maximum-length\",\n" +
+                "                                \"params\": {\n" +
+                "                                    \"maxLength\": 255\n" +
+                "                                },\n" +
+                "                                \"policyRequirements\": [\n" +
+                "                                    \"MAX_LENGTH\"\n" +
+                "                                ]\n" +
+                "                            }\n" +
+                "                        ],\n" +
+                "                        \"conditionalPolicies\": null\n" +
+                "                    }\n" +
+                "                },\n" +
+                "                {\n" +
+                "                    \"name\": \"failedPolicies\",\n" +
+                "                    \"value\": []\n" +
+                "                },\n" +
+                "                {\n" +
+                "                    \"name\": \"validateOnly\",\n" +
+                "                    \"value\": false\n" +
+                "                },\n" +
+                "                {\n" +
+                "                    \"name\": \"value\",\n" +
+                "                    \"value\": \"\"\n" +
+                "                }\n" +
+                "            ]\n" +
+                "        }");
+        StringAttributeInputCallback callback = new StringAttributeInputCallback(raw, 0);
+        Assertions.assertThat(callback.getName()).isEqualTo("sn");
+        Assertions.assertThat(callback.getPrompt()).isEqualTo("Last Name");
+        Assertions.assertThat(callback.isRequired()).isTrue();
+        Assertions.assertThat(callback.getPolicies().getString("name")).isEqualTo("sn");
+        Assertions.assertThat(callback.getFailedPolicies()).isEmpty();
+        Assertions.assertThat(callback.getValidateOnly()).isFalse();
+        Assertions.assertThat(callback.getValue()).isEmpty();
+
+
+    }
+}
```

Comment:
Do you think is valid to include more unit tests to test the input value, such as non-empty strings, very long strings, special characters, and so on?

Reply:
The test more focus on parsing the input, there is no special logic on assigning the input json, it just assigned it as is.